File: src/main/java/com/gmail/borlandlp/minigamesdtools/arena/localevent/ArenaPlayerDeathLocalEvent.java
package com.gmail.borlandlp.minigamesdtools.arena.localevent;

import org.bukkit.entity.Player;
import org.bukkit.event.Cancellable;

@ArenaEventHandler
public class ArenaPlayerDeathLocalEvent extends ArenaEvent implements Cancellable {
    private Player player;
    private boolean canceled;

    public ArenaPlayerDeathLocalEvent(Player player) {
        this.player = player;
    }

    public Player getPlayer() {
        return player;
    }
    @Override
    public boolean isCancelled() {
        return this.canceled;
    }

    @Override
    public void setCancelled(boolean b) {
        this.canceled = b;
    }

    public String toString() {
        return "{Event:" + this.getClass().getSimpleName() + " Player" + this.getPlayer() + "}";
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/arena/localevent/ArenaPlayerDeathLocalEvent.kt
package com.gmail.borlandlp.minigamesdtools.arena.localevent

import org.bukkit.entity.Player
import org.bukkit.event.Cancellable

@ArenaEventHandler
class ArenaPlayerDeathLocalEvent(val player: Player) : ArenaEvent(), Cancellable {
    private var canceled = false

    override fun isCancelled(): Boolean {
        return canceled
    }

    override fun setCancelled(b: Boolean) {
        canceled = b
    }

    override fun toString(): String {
        return "{Event:" + this.javaClass.simpleName + " Player" + player + "}"
    }

}